### ISSUE TITLE
Caught unexpected java.lang.NullPointerException during lookup operation.

### ISSUE THREAD

**Issue by ensonic** [qtc-de/remote-method-guesser#52]
```
java -jar target/rmg-4.4.1-jar-with-dependencies.jar enum 192.168.1.10 1099
[-] Caught unexpected java.lang.NullPointerException during lookup operation.
[-] Please report this to improve rmg :)
[-] StackTrace:
java.lang.NullPointerException: Cannot read the array length because "boundNames" is null
	at de.qtc.rmg.networking.RMIRegistryEndpoint.lookup(RMIRegistryEndpoint.java:142)
	at de.qtc.rmg.operations.Dispatcher.obtainBoundObjects(Dispatcher.java:101)
	at de.qtc.rmg.operations.Dispatcher.dispatchEnum(Dispatcher.java:498)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at de.qtc.rmg.operations.Operation.invoke(Operation.java:331)
	at de.qtc.rmg.Starter.main(Starter.java:26)
```

**Comment by qtc-de** [comment]
Hi @ensonic :wave: 

I think a similar issue was submitted recently, but got deleted before I could take a deeper look into it (#51). It was closed with the comment `user error`. Taking this into account together with your provided stacktrace, I think I got the issue:

The underlying problem is probably a connection issue. When *remote-method-guesser* is unable to connect to the target and a `ConnectException` is raised, it is handled by the following function:

```java
    public static void connectException(Exception e, String callName)
    {
        Throwable t = ExceptionHandler.getCause(e);

        if( t instanceof java.net.ConnectException ) {

            String message = t.getMessage();

            if( message.contains("Connection refused") )
                ExceptionHandler.connectionRefused(e, callName, "call");

            if( message.contains("Network is unreachable") )
                ExceptionHandler.networkUnreachable(e, callName, "call");

        } else {
            ExceptionHandler.unexpectedException(e, callName, "call", true);
        }
    }
```

In all cases, this function should actually close the program after printing an error message to the user. However, if the exception message is neither `Network is unreachable` nor `Connection refused`, the function simply returns without closing the program. This is unexpected and causes problems in later stages of the code.

The issue should be fixed in 801a8e9 and 4d83fc5. You will still obtain an error, but this time, the error reason should be displayed correctly.

**Comment by ensonic** [comment]
Thanks. Thats less confusing:
```
> java -jar target/rmg-5.0.0-jar-with-dependencies.jar enum 192.168.1.42 1099
[-] Caught ConnectException during list call.
[-] Exception message: Die Wartezeit für die Verbindung ist abgelaufen[-] Cannot continue from here.
```
